risc0/zkvm/src/prove/mod.rs

```diff
@@ -227,6 +227,13 @@ where
     }
 
     fn prove_session(&self, session: &Session) -> Result<SessionReceipt> {
```

Comment:
The `Prover` trait system lets us have pluggable provers, I would recommend having the Bonsai based prover be a implementation of that which we can register / fetch via get_prover(). Then in `default_prover()` check for the env var and swap to the BonsaiProver.